Library/Homebrew/rubocops/formula_desc_cop.rb

```diff
@@ -0,0 +1,75 @@
+require_relative "./extend/formula_cop"
+require_relative "../extend/string"
+
+module RuboCop
+  module Cop
+    module Homebrew
+      # This cop audits `desc` in Formulae
+      #
+      # - Checks for existence of `desc`
+      # - Checks if size of `desc` > 80
+      # - Checks if `desc` begins with an article
+      # - Checks for correct usage of `command-line` in `desc`
+      # - Checks if `desc` contains the formula name
+
+      class FormulaDesc < FormulaCop
+        def on_formula(node, class_node, _parent_class_node, body)
```

Comment:
Inside `FormulaCop` there's a method `on_class` which does the following sequentially

-  Checks if the file being processed is inside /Taps/Homebrew/
-  If so, check if current class is being inherited from `Formula` ( Thus making sure that its a Homebrew Formula)
-  Call `on_formula` with arguments below. (`on_formula` is not yet defined as we will define it inside the CustomCop which inherits from `FormulaCop`)
-  `node` is the RuboCop::AST::node instance of whole formula
-  `class_node` is the RuboCop::AST::node instance of formula's name
-  `_parent_class_node` is the RuboCop::AST::node instance of `Formula` 
-  `body` is the RubCop::AST::node for ever thing thats inside the Formula

`class_node`, `_parent_class_node`, `body` are children of `node` and may be we can omit passing `node` as the argument to `on_formula` as its redundant.